### DISCUSSION TITLE
issue: integration between react-dropzone & useFieldArray

### DISCUSSION THREAD

**Discussion by ItayTur** [react-hook-form/react-hook-form#9057] [Q&A] [answered]
### Version Number

7.35.0

### Codesandbox/Expo snack

https://codesandbox.io/s/react-hook-form-js-forked-ugjxq0?file=/src/App.js

### Steps to reproduce

1. get `control` out of `useFormHook`
2. add the `control` to the `useFieldArray`, and get the `fields`
3. set `replace` on the handler: `onDrop` of the `react-dropzone` hook: `useDropzone`
4. click the dropzone and select a file.
5. print `fields`
6. See error: `fields` objects are missing parameters: size, name etc...

### Expected behaviour

fields objects contain all of the file properties.

when viewing files by watch: `const files = form.watch(files)`. files do contain all file properteis, but doesn't have `id`

### What browsers are you seeing the problem on?

Chrome

### Relevant log output

```shell
files from watch: 


[
   {
      size: 100,
      name: 'file-name'
      path: 'some-path',
      type: 'some-type'
   }
]
```

fields:

```
[
  {
     id: 'some-id',
     path: 'some-path' 
  }
]
```

might be related to this https://github.com/react-hook-form/react-hook-form/issues/8717

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by leapful** [comment, marked as answer]
@ItayTur : This is not an issue from react-hook-form. Native Web API "File" class has all properties are read-only by default and also not enumerable thus it cannot be cloned via object assigning of react-hook-form.

For example:

```
{ ...new File() } => return an empty object {}
```

Please never use custom object as a form values, this is a best practice when working with form. In your use case, just manually clone File instance from "onDrop" callback into a plain object with all properties that you need then pass it to "replace".

Something like:

```
onDrop: (files) => {
   replace(files.map(f => ({ size: f.size, name: f.name, // any other prop })))
}
```
